### DISCUSSION TITLE
Request Body Blocks

### DISCUSSION THREAD

**Discussion by JuanScerri** [postmanlabs/postman-flows#376] [Ideas]
Hello. Thought I'd share my idea because I think it would help improve the efficiency of Postman Flows. Currently in Postman Flows there are a few blocks which allows us to create or group data. However, I think it would be better if a Request Body block or a JSON block is added.

This block would behave in exactly the same way a record block behaves. However, instead of having to manually specify each key/value pair using a GUI which can be time consuming instead we can just copy and past some JSON into this block and Postman parses the JSON and provides a similar record interface for use with other blocks.

Moreover, I think it would also be beneficial if the Send Request block provides the possibility to completely override the request body with one of these blocks. This ensures that you do not need to design requests with Postman Flows in mind. I think this can be especially difficult to do when an API has a very large Body and you need control over each individual field.

**Comment by angels7777** [comment]
What for

**Reply by JuanScerri** [reply]
With the current system you cannot have a request be both a part of collection and used inside of the collection (for example running the collection) and in a flow at the same time. This is because you need to make sure that you embedded variables in the body of the request to properly use it in flows.

![image](https://user-images.githubusercontent.com/111490976/185374318-2ea84f41-dc2f-4f8d-8ca4-b6912b466b86.png)

I could try to solve this issue by specifying a variable in the collection itself however, many of the requests in the collection have different bodies hence it would result in a lot of variables and that does not make sense.

Is there a correct way of dealing with this issue?

**Comment by angels7777** [comment]
What's going on

**Comment by Mvrlex** [comment]
We are currently trying to implement some regression tests for our api in Postman Flows, and with the current workflow we would need to create a request for every varation of body we have. Which is a bit ugly.
My suggestion would be just allow selection of a predefined example when making a request, or something similar.

**Reply by JuanScerri** [reply]
Hello @Mvrlex thank you for your reply. I am looking forward to seeing these fixes come to Postman. Could you kindly give this discussion a look https://github.com/postmanlabs/postman-flows/discussions/248. I posted a video demonstrating some strange behavior with Postman flows as part of the discussion.
